### PR TITLE
[Form] collect all transformation failures

### LINKED ISSUE #37262: [Form] BC break in validation
This test passes with 3.4.40 but fails with 3.4.41 due to changes in #36865. (It also fails with 3.4.42.)

(I put it into `DateTypeTest` with some boilerplate code for the validator to easily switch between versions since the validator tests were moved between 3.4.41 and 3.4.42.)

I'm not sure if only DateType is affected. Just noticed this behavioral change on it.

/ping @xabbuh

```suggest
diff --git Tests/Extension/Core/Type/DateTypeTest.php Tests/Extension/Core/Type/DateTypeTest.php
index cb2c2d0..a568c78 100644
--- Tests/Extension/Core/Type/DateTypeTest.php
+++ Tests/Extension/Core/Type/DateTypeTest.php
@@ -16,6 +16,12 @@
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Intl\Util\IntlTestHelper;
 
+use Symfony\Component\Form\Extension\Validator\ValidatorExtension;
+use Symfony\Component\Form\Forms;
+use Symfony\Component\Validator\Mapping\Factory\LazyLoadingMetadataFactory;
+use Symfony\Component\Validator\Mapping\Loader\StaticMethodLoader;
+use Symfony\Component\Validator\Validation;
+
 class DateTypeTest extends BaseTypeTest
 {
     const TESTED_TYPE = 'Symfony\Component\Form\Extension\Core\Type\DateType';
@@ -220,6 +226,33 @@
         $this->assertEquals($text, $form->getViewData());
     }
 
+    public function testSubmitFormChoiceInvalid()
+    {
+        $this->validator = Validation::createValidatorBuilder()
+            ->setMetadataFactory(new LazyLoadingMetadataFactory(new StaticMethodLoader()))
+            ->getValidator();
+
+        $this->factory = Forms::createFormFactoryBuilder()
+            ->addExtension(new ValidatorExtension($this->validator))
+            ->getFormFactory();
+
+        $form = $this->factory->create(static::TESTED_TYPE, null, [
+            'widget' => 'choice',
+            'years' => [2021],
+        ]);
+
+        $form->submit([
+            'day' => '13',
+            'month' => '6',
+            'year' => '2020',
+        ]);
+
+        $this->assertTrue($form->isSubmitted());
+        $this->assertFalse($form->isValid());
+        $this->assertCount(1, $form->getErrors());
+        $this->assertSame('This value is not valid.', $form->getErrors()->current()->getMessage());
+    }
+
     public function testSubmitFromChoiceEmpty()
     {
         $form = $this->factory->create(static::TESTED_TYPE, null, [
```

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37262
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php [hunk 1]
```diff
@@ -329,6 +332,61 @@ public function testContextIsPopulatedWithFormBeingValidatedUsingGroupSequence()
...
+            'day' => '13',
+            'month' => '13',
+            'year' => '2020',
+        ]);
+
+        $this->assertTrue($form->isSubmitted());
+        $this->assertFalse($form->isValid());
+        $this->assertCount(2, $form->getErrors());
+        $this->assertSame('This value is not valid.', $form->getErrors()[0]->getMessage());
+        $this->assertSame('This value is not valid.', $form->getErrors()[1]->getMessage());
```
Comment (craue) [inline src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:353]: I wonder why this error is added twice.

Comment (xabbuh) [reply src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:353]: Once for the month and once for the year

Comment (xabbuh) [reply src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:353]: The removed `continue` is responsible for it. Maybe we should keep it?

Comment (craue) [reply src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:353]: Hah, true. I didn't notice the invalid month. Was focused on the year only due to my initial test. 😄

Comment (craue) [reply src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:353]: Should the cause of the error be tested as well then? To ensure one is really caused by the month and one by the year.

Comment (craue) [reply src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php:353]: 3.4.40 also adds two errors in this case so it should be fine to keep that behavior.

Comment (xabbuh) [reply src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorFunctionalTest.php]: > Should the cause of the error be tested as well then? To ensure one is really caused by the month and one by the year.

Yes, totally makes sense. I did that.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**